client/landing/gutenboarding/hooks/use-detect-matching-anchor-site.ts

```diff
@@ -34,22 +33,23 @@ export default function useDetectMatchingAnchorSite(): boolean {
 		}
 
 		setIsLoading( true );
-		wpcom
-			.undocumented()
-			.getMatchingAnchorSite(
-				anchorFmPodcastId,
-				anchorFmEpisodeId,
-				anchorFmSpotifyUrl,
-				anchorFmSite,
-				anchorFmPost
-			)
+
+		// construct query object from entries that are not null
+		const query = Object.fromEntries(
+			[
+				[ 'podcast', anchorFmPodcastId ],
+				[ 'episode', anchorFmEpisodeId ],
+				[ 'spotify_url', anchorFmSpotifyUrl ],
+				[ 'site', anchorFmSite ],
+				[ 'post', anchorFmPost ],
+			].filter( ( [ , value ] ) => value != null )
```

Comment:
Just to confirm that we're intentionally also filtering out `undefined` values here, while previously we used to filter out only `null` ones`. This is [typed](https://github.com/Automattic/wp-calypso/blob/migrate-wpcom-undoc-anchor/client/landing/gutenboarding/path.ts#L137) to be a string or null, so getting an `undefined` is very unlikely, but still, in that case, should we make the check a strict one?

Reply:
The undefined query values will be filtered out by the `wpcom` library itself. I tested this behavior while working on the PR:
```
wpcom.req.get( '/me', { foo: undefined, bar: null } )
```
will send a request with query string:
```
/me?bar=
```
`undefined` is filtered out by the library, `null` is not.

Reply:
In that case that's an improvement. If at any point in the future we decide to use a different request library, filtering out `undefined` values explicitly makes more sense than relying on the request library to do it for us 👍